pkg/operator/options/suite_test.go

```diff
@@ -115,6 +116,21 @@ var _ = Describe("Options", func() {
 	})
 
 	Context("Validation", func() {
+		It("should fail validation when networkPluginMode is invalid", func() {
+			typo := "overlaay"
+			errMsg := fmt.Sprintf("network-plugin-mode %v is invalid. network-plugin-mode must equal 'overlay', 'none' or ''", typo)
+
+			err := opts.Parse(
+				fs,
+				"--cluster-name", "my-name",
+				"--cluster-endpoint", "https://karpenter-000000000000.hcp.westus2.staging.azmk8s.io",
+				"--kubelet-bootstrap-token", "flag-bootstrap-token",
+				"--ssh-public-key", "flag-ssh-public-key",
+				"--vm-memory-overhead-percent", "-0.01",
+				"--network-plugin-mode", typo,
+			)
+			Expect(err).To(MatchError(ContainSubstring(errMsg)))
+		})
```

Comment:
These tests are brittle as the validation error message is dependent on the ordering of the calls and what flags we pass in. There is probably a better way with multi-error to test that we saw the error we are looking for.